charmcraft/commands/init.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2020 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+from datetime import date
+import logging
+import os
+from pathlib import Path
+import pwd
+import re
+
+from jinja2 import Environment, PackageLoader, StrictUndefined
+
+from charmcraft.cmdbase import BaseCommand, CommandError
+from ._utils import make_executable
+
+logger = logging.getLogger(__name__)
+
+
+class InitCommand(BaseCommand):
+    """Initialize a directory to be a charm project."""
+    name = "init"
+    help_msg = "initialize a directory to be a charm project"
+
+    def fill_parser(self, parser):
+        parser.add_argument(
+            "-d", "--directory", type=Path, default=Path("."),
```

Comment:
`--directory` is too ambiguous... yes, it's a directory, a directory for what? Maybe `--base-directory`, or `--base`, or `--deploy-directory`?

Reply:
both of those are worse, for different reasons: 'base' doesn't add anything other than length (ie it doesn't _mean_ anything), and 'deploy' means something else entirely :-)
but, yes, directory is not clear.

Reply:
is `--project-directory` too long? Maybe `--project-dir`?

Reply:
I'm liking `--project-dir`, but it does open the question as to whether `build` shouldn't use the same name for the same thing. And if not, maybe `init` should use `--to` (cf. `build`'s `--from`), but that's less than clear for the user imho.